Library/Homebrew/formula_installer.rb

```diff
@@ -343,6 +344,19 @@ def check_install_sanity
           "#{formula.full_name} requires the latest version of pinned dependencies"
   end
 
+  sig { void }
+  def install_fetch_deps
+    return if @compute_dependencies.blank?
+
+    @compute_dependencies.delete_if do |dep, options|
+      if dep.tags.include? :fetch
+        fetch_dependency(dep)
+        install_dependency(dep, options)
+        next true
+      end
```

Comment:
```suggestion
      next false unless dep.tags.include? :fetch
      
      fetch_dependency(dep)
      install_dependency(dep, options)
      true
```

Reply:
Used this with `any?` to solve the "installation already attempted" issue by having it recompute dependencies if any `:fetch` deps were installed.